packages/video_player/video_player/lib/video_player.dart

```diff
@@ -888,12 +887,15 @@ class _VideoPlayerWithRotation extends StatelessWidget {
   final Widget child;
 
   @override
-  Widget build(BuildContext context) => rotation == 0
-      ? child
-      : Transform.rotate(
-          angle: rotation * math.pi / 180,
-          child: child,
-        );
+  Widget build(BuildContext context) {
+    if (rotation % 90 != 0 || rotation == 0) {
```

Comment:
Why is silently ignoring any rotation value that's not a multiple of 90 correct? If there are actually cases where this happens, wouldn't silently ignoring them be even more broken than the current behavior?

Reply:
@stuartmorgan  
The `rotation` value in this context can only be 0, 90, 180, or 270 degrees, as most video formats (such as MP4 and MOV) store rotation metadata in **90-degree increments** within the `tkhd` (Track Header) box.  

For example, in the current Flutter ExoPlayer backend, any value outside this range is ignored:  
https://github.com/flutter/packages/blob/2d3b24ebd9e8d72399d1176089065440e18cc033/packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/texture/TextureExoPlayerEventListener.java#L67-L74  

Since `RotatedBox` only supports quarter-turn rotations, this shouldn't be an issue. However, if an unexpected rotation value appears, should we fall back to `Transform.rotate` instead of ignoring it?

Reply:
If the expectation is that it's impossible for this Dart code to receive values that aren't multiples of 90 degrees, that should be expressed via an assert, not a runtime conditional.

Reply:
I've added an assert to ensure that only multiples of 90 degrees are allowed.

Reply:
I've updated the conditional statement to reflect the assert change